pkg/operation/botanist/kubeapiserver.go

```diff
@@ -198,29 +202,56 @@ func (b *Botanist) ensureAdmissionPluginConfig(plugins []gardencorev1beta1.Admis
 		}
 	}
 
+	if index == -1 {
+		return plugins, nil
+	}
+
 	// If user has set a config in the shoot spec, retrieve it
 	if plugins[index].Config != nil {
+		var (
+			admissionConfigData []byte
+			err                 error
+		)
+
 		config, err := runtime.Decode(codec, plugins[index].Config.Raw)
 		if err != nil {
 			return nil, err
 		}
-		admissionConfig, ok := config.(*admissionapiv1beta1.PodSecurityConfiguration)
-		if !ok {
-			return nil, fmt.Errorf("expected admissionapiv1beta1.PodSecurityConfiguration but got %T", config)
-		}
 
-		// Add kube-system to exempted namespaces
-		if !slices.Contains(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem) {
-			admissionConfig.Exemptions.Namespaces = append(admissionConfig.Exemptions.Namespaces, metav1.NamespaceSystem)
-		}
+		if versionutils.ConstraintK8sGreaterEqual123.Check(b.Shoot.KubernetesVersion) {
```

Comment:
Is `v1alpha1` removed in 1.23+? Or why do we enforce that `v1beta1` is used here? Shouldn't we take whatever the user provides and be fine with it?

Reply:
Fair enough, but `v1beta1` config shouldn't be passed to clusters below v1.23, right?

Reply:
Yeah, for now let's not get into the business of validating these configurations. :)